doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
+ * the field of scalars ``F`` are the real numbers.
+
+The first assumption is made in order to simplify the concept of *convex conjugate functional*, see ``conjugate_functional`` under :ref:`implementation` for more details, or the Wikipedia articles on `convex conjugates`_ and `Legendre transformations`_.
+
+The second assumption is made in order to guarantee that the optimization problem, in some way, makes sense.
+For example, if the field is not (totally) ordered (like for example the complex numbers) one cannot define optimization over it in a meaningful way.
+Or if the field does not have the least-upper-bound property, then attainment of an optimal solution cannot be guaranteed.
+See, for example, the Wikipedia articles on `field`_, `ordered field`_ and `least-upper-bound property`_.
+
+Note that it is possible to use the class without fulfilling these assumtptions, however in this case some of the mathematical results might not hold.
```

Comment:
"undefined behavior" in programming generally means ANYTHING CAN HAPPEN. In c++ formatting the hard-drive on undefined behavior is considered a perfectly valid response. I prefer the weaker "unknown" behavior, or simply "bugs".